controllers/routemonitor/routemonitor_controller_supplement.go

```diff
@@ -168,16 +167,203 @@ func (r *RouteMonitorReconciler) CreateBlackBoxExporterService(ctx context.Conte
 	}
 	return nil
 }
-func (r *RouteMonitorReconciler) CreateBlackBoxExporterResources(ctx context.Context) error {
-	if err := r.CreateBlackBoxExporterDeployment(ctx); err != nil {
+
+func (r *RouteMonitorReconciler) CreateServiceMonitorResource(ctx context.Context, routeMonitor *v1alpha1.RouteMonitor) (*ctrl.Result, error) {
+	// Was the RouteURL populated by a previous step?
+	if routeMonitor.Status.RouteURL == "" {
+		return nil, customerrors.NoHost
+	}
+
+	if !r.HasFinalizer(routeMonitor) {
+		// If the routeMonitor doesn't have a finalizer, add it
+		utilfinalizer.Add(routeMonitor, FinalizerKey)
+		if err := r.Update(ctx, routeMonitor); err != nil {
+			return nil, err
+		}
+		// After any modification we need to requeue to prevent two threads working on the same code
+		return &ctrl.Result{Requeue: true}, nil
+	}
+
+	namespacedName := r.templateForServiceMonitorName(routeMonitor)
+
+	resource := &monitoringv1.ServiceMonitor{}
+	populationFunc := func() *monitoringv1.ServiceMonitor {
+		return r.templateForServiceMonitorResource(routeMonitor, namespacedName.Name)
+	}
+
+	// Does the resource already exist?
+	if err := r.Get(ctx, namespacedName, resource); err != nil {
+		// If this is an unknown error
+		if !k8serrors.IsNotFound(err) {
+			// return unexpectedly
+			return nil, err
+		}
+		// populate the resource with the template
+		resource := populationFunc()
+		// and create it
+		err = r.Create(ctx, resource)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return nil, nil
+}
+
+/*
+
+This next block is for deletion of the RouteMonitor
+
+*/
```

Comment:
I think function comments are required to look different in according to [effective go](https://golang.org/doc/effective_go.html#commentary)

Reply:
yeah, will remove this function as it served convenience for order in my head